server/rest/src/main/java/org/infinispan/rest/framework/impl/SimpleRequest.java

```diff
@@ -0,0 +1,185 @@
+package org.infinispan.rest.framework.impl;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Map;
+
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.rest.framework.ContentSource;
+import org.infinispan.rest.framework.Method;
+import org.infinispan.rest.framework.RestRequest;
+
+import io.netty.handler.codec.http.QueryStringDecoder;
+
+/**
+ * @since 10.0
+ */
+public class SimpleRequest implements RestRequest {
+   private final Method method;
+   private final String path;
+   private Map<String, String> headers;
```

Comment:
`headers` is never used

Reply:
```SimpleRequest``` is the default implementation of ```RestRequest``` not tied to any particular underlying http library, this allows to test the base frameworks classes in isolation. As it evolves, I expect it to the used in future tests